build.gradle

```diff
@@ -91,18 +90,11 @@ allprojects {
 
     gradle.projectsEvaluated {
         tasks.withType(JavaCompile) {
-            sourceCompatibility = JavaVersion.VERSION_1_8
-            targetCompatibility = JavaVersion.VERSION_1_8
+            sourceCompatibility = JavaVersion.VERSION_11
+            targetCompatibility = JavaVersion.VERSION_11
```

Comment:
Is there a way to override this for the client?

Reply:
I don't believe it is sufficient (or even possible) to supply overrides for the client (and its dependencies). For example if I supply 

```
    tasks.withType(JavaCompile) {
        sourceCompatibility = JavaVersion.VERSION_1_8
        targetCompatibility = JavaVersion.VERSION_1_8
    }
```

in any other project, it still builds Java 11 classes. Even if this was just a matter of ordering this configuration, I don't this it is sufficient. See: https://bugs.openjdk.java.net/browse/JDK-8058150

The way we currently override it is through the `--release` flag, which basically does the above anyway.

Reply:
My understanding is that `--release` is not an equivalent of `--source` and `--target` , you should always use `--release` starting with Java9+. With that in mind, do we really need to specify `--source` and `--target` explicitly for the components we want to build with the current JDK (e.g. JDK11)?

For Client and its dependencies we should use `--release` explicitly as we do now.

Reply:
We don't need to explicitly include them no. I do however think it is a good idea to also include release flags for the non-client components. This should protect any users who are running some JDK version >11 from running into issues.